### ISSUE TITLE
Pylance not working in vs code interactive window

### ISSUE THREAD

**Issue by dataagile-ema** [microsoft/pylance-release#1586]
### Discussed in https://github.com/microsoft/pylance-release/discussions/1585

<div type='discussions-op-text'>

<sup>Originally posted by **dataagile-ema** July 22, 2021</sup>
Hi!

Whenever pylance detects a problem when i am in the interactive window in vs code, it just stops working. It no longer gives autocomplete (to make things worse, pylance detects problem even when there is no problem). I hope the attached image shows the problem.
I would greatly appreciate help on ths, it i really irritating. Pylance autocomplete really speads up development.

![vs code pylance](https://user-images.githubusercontent.com/25013783/126618036-fd24eaee-d4b1-4cee-ac3e-0d365d9adba4.png)

</div>

**Comment by judej** [comment]
Would you please provide logs and repro steps as described in the [troubleshooting guide](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md), please?

**Comment by dataagile-ema** [comment]
pylance version v2021.7.5
[g_environment_yml.txt](https://github.com/microsoft/pylance-release/files/6870531/g_environment_yml.txt)
'Python 3.9.6 64-bit

The code is pretty much what you see in the image I attached, but the code could be anything, e.g:
import pandas as pd
import altair as alt
df = pd. does give any suggestions for methods in pandas module. Sometimes autocomplete works for a short while, then it stops.
I set VS Code tol loggning Trace, but the the debud console does not output anything. 
Hope that helps.

**Comment by jakebailey** [comment]
The logs we are interested in are printed in the Python Language Server output window, here:

![image](https://user-images.githubusercontent.com/5341706/126826112-935b3348-a869-487a-970f-9aac6360c040.png)

**Comment by dataagile-ema** [comment]
[langserverlog.txt](https://github.com/microsoft/pylance-release/files/6870597/langserverlog.txt)
here is the log

**Comment by jakebailey** [comment]
Things aren't working because the LS is crashing.

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received null
    at validateString (internal/validators.js:124:11)
    at Object.basename (path.js:671:5)
    at Object.getFileName (c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\pyright-internal\src\common\pathUtils.ts:535:17)
    at listener (c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\pyright-internal\src\analyzer\service.ts:1282:38)
    at FSWatcher.<anonymous> (c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\pyright-internal\src\common\realFileSystem.ts:387:25)
    at FSWatcher.emit (events.js:315:20)
    at FSEvent.FSWatcher._handle.onchange (internal/fs/watchers.js:186:12)
```

The logs show that one of your paths is apparently empty:

```
Search paths for 
  c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\typeshed-fallback\stdlib
  
  typings
  c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\typeshed-fallback\stubs\...
  c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\bundled\stubs
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\DLLs
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages\win32
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages\win32\lib
  c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages\pythonwin
```

Potentially null. I think this is because you are not opening a folder (open file mode has a few known issues), but we shouldn't be crashing or adding an empty workspace path as a search path (or then try to watch it).

**Comment by dataagile-ema** [comment]
> 
> 
> Things aren't working because the LS is crashing.
> 
> ```
> TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received null
>     at validateString (internal/validators.js:124:11)
>     at Object.basename (path.js:671:5)
>     at Object.getFileName (c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\pyright-internal\src\common\pathUtils.ts:535:17)
>     at listener (c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\pyright-internal\src\analyzer\service.ts:1282:38)
>     at FSWatcher.<anonymous> (c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\pyright-internal\src\common\realFileSystem.ts:387:25)
>     at FSWatcher.emit (events.js:315:20)
>     at FSEvent.FSWatcher._handle.onchange (internal/fs/watchers.js:186:12)
> [Info  - 20:14:54] Connection to server got closed. Server will restart.
> [Info  - 20:14:55] Pylance language server 2021.7.5 (pyright 06d75c6c) starting
> ```
> 
> The logs show that one of your paths is apparently empty:
> 
> ```
> Search paths for 
>   c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\typeshed-fallback\stdlib
>   
>   typings
>   c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\typeshed-fallback\stubs\...
>   c:\Users\nikla.DESKTOP-78VQUGU\.vscode\extensions\ms-python.vscode-pylance-2021.7.5\dist\bundled\stubs
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\DLLs
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages\win32
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages\win32\lib
>   c:\Users\nikla.DESKTOP-78VQUGU\Anaconda3\Lib\site-packages\pythonwin
> ```
> 
> Potentially null. I think this is because you are not opening a folder (open file mode has a few known issues), but we shouldn't be crashing or adding an empty workspace path as a search path (or then try to watch it).

I could do another log, when actually opening an repo and then going into interactive window. I am pretty sure that was what I did when I posted the problem yesterday (see original post and image). Would it be of help to have that log too?

**Comment by jakebailey** [comment]
I think we have enough info.

You should be able to work around this by using `Open Folder`, rather than just opening up your notebook file. Can you confirm that it works when you do that?

**Comment by dataagile-ema** [comment]
First the workaround  works, i get the autocomplete as expeted, and then i stops, see image

![image_vscode](https://user-images.githubusercontent.com/25013783/126828847-cedc0386-1746-432d-8477-a3a6a0a2467f.png)

[langserverlog_part2.txt](https://github.com/microsoft/pylance-release/files/6870678/langserverlog_part2.txt)

**Comment by jakebailey** [comment]
The other thing to confirm is that the interpreter that you've selected is the same one that is running the kernel. Pylance uses the interpreter that the Python extension chooses, but Jupyter uses a separate selection that isn't always synchronized.

Try creating a plain python file in that folder and write the same code, and see if `alt` works.

**Comment by dataagile-ema** [comment]
When a create a plain python file, everything works, a get autcomplete for alt. And that is my experience, I have this problem when running interactive window.

**Comment by psobolewskiPhD** [comment]
Same issues here in Version: 1.59.0-insider (apple silicon native)
Really annoying!

**Comment by jakebailey** [comment]
The next version should fix various errors / bad behaviors caused by open-file mode; we weren't being too careful about handling an unrooted execution environment in a number of places (but now handle it explicitly).

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202192-16-september-2021
